### ISSUE TITLE
Error converting loom from HCA

### ISSUE THREAD

**Issue by dagarfield** [satijalab/seurat#2764]
I'm attempting to do a few analyses on the (rather large) immune cell atlas from the HCA:
https://data.humancellatlas.org/explore/projects/cc95ff89-2e68-4a08-a234-480eca21ce79

The data matrix can be downloaded in loom format. This seems to work at the start. 
>library(loomR)
library(Seurat)
>lfile <- connect(filename = "immuneCells.loom", mode = "r+")
> lfile
Class: loom
Filename: /gpfs01/home/glimi/local_projects/processing_scImmuneCells/data/alt_download/loom_version/immuneCells.loom
Access type: H5F_ACC_RDWR
Attributes: CreationDate, LOOM_SPEC_VERSION, last_modified, version
Listing:
       name    obj_type   dataset.dims dataset.type_class
  col_attrs   H5I_GROUP           <NA>               <NA>
 col_graphs   H5I_GROUP           <NA>               <NA>
     layers   H5I_GROUP           <NA>               <NA>
     matrix H5I_DATASET 782859 x 58347          H5T_FLOAT
  row_attrs   H5I_GROUP           <NA>               <NA>
 row_graphs   H5I_GROUP           <NA>               <NA>

But things go sideways at the conversion step
>immuneData<-as.Seurat(lfile)
Pulling counts matrix |                                                  | 0 % ~calculating  Error in self[[dataset]][chunk.indices, ][indices.use, features] :  incorrect number of dimensions

Suggestions on what I might try?

PS: It seems clear that a smarter alternative might be to use the loom branch of Seurat instead -- this is many cells in any case. I'll try that in parallel.

**Comment by dagarfield** [comment]
Aaaaand put a fork in this one. Simple answer: In this dataset, the Cell IDs and Gene Names were misnamed, and it just had to be specified properly in the command....

**Comment by gbloeb** [comment]
@dagarfield Can you explain how you did this.  Struggling with same issue for another HCA dataset

**Comment by dagarfield** [comment]
In the end, I used the loom file from the Broad repository, so I don't have my direct HCA download anymore. However, two things to check. One is to take a quick tour through the loomR tutorial (https://satijalab.org/loomR/loomR_tutorial.html) where they walk you through how to check the names of the colData/rowData entries for your cell ids and gene ids [Seurat assumes cells = "CellID" and features = "Gene", but this is not universal]. The other is to try and simply build the Seurat object directly from the loom file. To read the matrix, I used 'scrattch.io::read_loom_dgCMatrix' rather than trying to import a monster matrix via loomR. But the rest of the required data is in 'col_attrs' and can be imported pretty easily.
